src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -455,16 +455,11 @@ private static IEnumerable<Decorator> GetSystemDecorators()
                 .WithRequiredParameter("values", LanguageConstants.Array, "The allowed values.")
                 .WithFlags(FunctionFlags.ParameterDecorator)
                 .WithValidator((_, decoratorSyntax, targetType, typeManager, diagnosticWriter) =>
-                 {
-                     // The type checker should already verified that there's only one array argument.
-                     var allowedValuesArray = (ArraySyntax)SingleArgumentSelector(decoratorSyntax);
```

Comment:
>ArraySyntax [](start = 47, length = 11)

The cast can fail in certain cases. (See #1713.)

Reply:
Good catch!